src/roswire/ros2/node.py

```diff
@@ -2,28 +2,61 @@
 __all__ = ('ROS2Node',)
 
 import typing
-
 import attr
 
+from loguru import logger
+import dockerblade
+
+from ..interface import Node
+from ..exceptions import NodeShutdownError
+from .state import ROS2StateProbe
+
 if typing.TYPE_CHECKING:
     from ..app import AppInstance
 
 
 @attr.s(frozen=True, auto_attribs=True, slots=True)
-class ROS2Node:
+class ROS2Node(Node):
     """Provides an interface for interacting with a ROS2 node.
 
     Attributes
     ----------
     name: str
         The name of the node.
     """
-    app_instance: 'AppInstance'
+    app_instance: 'AppInstance' = attr.ib()
```

Comment:
Since `auto_attribs` is being used, we don't need to explicitly state `= attr.ib()` for `app_instance` or `name`.